proposals/005_Experimental.md

```diff
@@ -0,0 +1,115 @@
+# Experimental marker
+
+## Metadata
+
+|Tag |Value |
+|---- | ---------------- |
+|Proposal |[005_Experimental](https://github.com/OAI/OpenAPI-Specification/tree/master/proposals/005_Experimental.md)|
+|Authors|[David Goss](https://github.com/davidjgoss)|
+|Review Manager |TBD |
+|Status |Proposal|
+|Implementations ||
+|Issues ||
+|Previous Revisions ||
+
+## Change Log
+
+|Date |Responsible Party |Description |
+|---- | ---------------- | ---------- |
+
+## Introduction
+
+A way to mark an aspect of the API as "experimental", indicating that it is not yet a fully stable and supported part of the API.
+
+## Motivation
+
+Consider an API with two categories of thing in it:
+
+- Core, stable things, where we are committed to the ongoing stability and have no intention of making breaking changes.
+- New, experimental things, where we are getting them out there for feedback and early adopters, but they may change before we consider them to be in the first category, or even just get removed.
+
+These sit together fine in principle, but cause friction when trying to apply something like semver to the API as a whole. How do we make changes to the experimental stuff - without bumping the major version several times a year and scaring consumers - while also ensuring we can't make breaking changes to the core stuff we never _want_ to break.
+
+## Proposed solution
+
+Add an "experimental" field which specifies that an items in the API is not yet fully stable and supported, may change or be removed without a major version bump, and as such should be used with caution.
+
+_(I don't have a strong opinion about the naming - "beta" is another idea, though I think "experimental" does the job better in terms of being the most noncommital.)_
+
+Downstream tools could then make use of this metadata:
+
+- Tools like swagger-ui could surface this in the documentation they generate so consumers are made aware. Experimental items could also be filtered out of the documentation if desired.
+- Tools for detecting and preventing breaking changes could take this into consideration when deciding whether a change is breaking.
+
+## Detailed design
+
+A new boolean field named `experimental`, defaulting to `experimental`, is added to:
+
+- Operation
+- Schema
```

Comment:
I wonder whether adding the `experimental` field to higher level objects (such as `parameter`, `requestBody` and `response`) and dropping it from `schema` would work better? It might resolve the 'unanswered question' about experimental schema use in non-experimental operations.

Reply:
I'd missed parameter - have added that now, with a qualifier that it couldn't be used with `in: path` since a change in path is a breaking change.

Reply:
I see your point about `schema`, I agree removing it from there would simplify. I'm not sure I understand how putting on `requestBody` and `response` instead would alleviate though, could you elaborate?

One use case I've had come up is having a subset of an object be experimental, like "we've added some more detail to this response, we're not totally sure about it yet but the rest of it remains stable in the meantime", so it would be a shame to lose that.